crates/gpui/src/platform/linux/platform.rs

```diff
@@ -391,15 +391,51 @@ impl Platform for LinuxPlatform {
     }
 
     fn write_credentials(&self, url: &str, username: &str, password: &[u8]) -> Task<Result<()>> {
-        unimplemented!()
+        let url = url.to_string();
+        let username = username.to_string();
+        let password = password.to_vec();
+        self.background_executor().spawn(async move {
+            let keyring = oo7::Keyring::new().await?;
+            keyring.unlock().await?;
+            keyring.create_item(&url, &vec![("url", &url), ("username", &username)], password, true).await?;
```

Comment:
I think we should probably use our own label, right? Something like `zed-github-login`

Reply:
I hesitated to use something zed related, because gpui will (and already is) used by other applications as well. I thought about using the platform's path api to find an application name but it seemed unnecessary complex, so I ended up just giving no label at all. In the future we may want to expose the label in the creds api, but I guess your approach works just as well for now. Happy it got merged :)